docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,55 @@
+= Proposal: Multi build repo skipping
+
+== Purpose
+
+Skip builds for which nothing has changed in their working directory.
+
+== Background
+
+For a repo with a single build, the pipeline build will only be triggered if any changes trigger a new pipeline run. 
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed. 
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task. 
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.  
+
+== Solution
+
+`ods-start` provides a file called `$WORKING_DIR/.ods/git-dir-commit-sha` which contains the git commit sha of the working directory (no whitespace or newlines).
```

Comment:
I am not sure I understand how this works.

At the moment, there `ods-start` checks out the repo, and create a `$CHECKOUT_DIR/.ods/git-commit-sha` file.

`ods-start` has no concept of a "working dir" so far. Each build task can specify one, but there would be no file inside the working dir which contains the Git commit SHA.

Could you describe how you envision this to work?

Reply:
We probably do not want to have other tasks needing `git` in their images and leave git things to `ods-start`. So `ods-start` could by convention look for `working-dir` parameters in other tasks and for each of these do a `working_dir_commit_sha=$(git rev-parse "HEAD:$working_dir")` and then write the file as described (see https://stackoverflow.com/a/58600111). 
I tested this a bit locally as I am using https://gist.github.com/henrjk/4265bc1b4f6fe3f7cd9027b0bcc284f9 to test task caching stuff before. At least in that context this appeared to work for me.

Reply:
> ods-start could by convention look for working-dir parameters in other tasks

In ods-start this would require API calls I believe to retrieve the pipeline. Another place could be the webhook interceptor, which has direct access to other tasks.

Also, I checked whether Git is currently available in the tasks. For Go/TypeScript it is. For Python/Gradle it isn't ... I'd say we could at least consider installing git everywhere if that makes things considerably easier.